### ISSUE TITLE
Cannot read property 'exports' of undefined

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#36983]
**TypeScript Version:**  3.9.0-dev.20200223

**Search Terms:**
- updateOpen
- applyChangesInOpenFiles

**Repo**

1. `git clone https://github.com/thanpolas/middlewarify.git`
1. Open `index.js` in  VSCode

**Bug**

```
Cannot read property 'exports' of undefined
TypeError: Cannot read property 'exports' of undefined
    at declareModuleMember (tsserver.js:30420:63)
    at delayedBindJSDocTypedefTag (tsserver.js:31843:29)
    at bindSourceFile (tsserver.js:30121:17)
    at Object.bindSourceFile (tsserver.js:30054:9)
    at initializeTypeChecker (tsserver.js:66453:20)
    at Object.createTypeChecker (tsserver.js:34861:9)
    at Object.getTypeChecker (tsserver.js:98607:79)
    at synchronizeHostData (tsserver.js:135685:21)
    at Proxy.getProgram (tsserver.js:135767:13)
    at InferredProject.Project.updateGraphWorker (tsserver.js:140244:53)
    at InferredProject.Project.updateGraph (tsserver.js:140187:42)
    at ProjectService.assignOrphanScriptInfoToInferredProject (tsserver.js:142163:25)
    at ProjectService.assignProjectToOpenedScriptInfo (tsserver.js:143593:26)
    at tsserver.js:143858:107
    at Object.mapDefined (tsserver.js:562:30)
    at ProjectService.applyChangesInOpenFiles (tsserver.js:143858:48)
    at Session.handlers.ts.createMapFromTemplate._a.<computed> (tsserver.js:144722:46)
    at tsserver.js:146465:88
    at IOSession.Session.executeWithRequestId (tsserver.js:146456:28)
    at IOSession.Session.executeCommand (tsserver.js:146465:33)
    at IOSession.Session.onMessage (tsserver.js:146489:35)
    at Interface.<anonymous> (tsserver.js:147804:27)
    at Interface.emit (events.js:203:13)
    at Interface._onLine (readline.js:316:10)
    at Interface._normalWrite (readline.js:461:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:203:13)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:276:11)
    at Socket.Readable.push (_stream_readable.js:210:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:166:17) 
```

**Comment by mjbvz** [comment]
@RyanCavanaugh Marking as priority since this causes VS Code to restart the TS Server. We'd like a fix for a TS 3.8.x recovery

**Comment by RyanCavanaugh** [comment]
Compile with `--allowJs`
```
var middlewarify = module.exports = {};

/** @enum */
middlewarify.Type = {
  BEFORE: 'before'
};
```

```
                        return declareSymbol(container.symbol.exports, container.symbol, node, symbolFlags, symbolExcludes); // No local symbol for an unnamed default!
                                                              ^

TypeError: Cannot read property 'exports' of undefined
    at declareModuleMember (D:\github\TypeScript\built\local\tsc.js:30390:63)
    at delayedBindJSDocTypedefTag (D:\github\TypeScript\built\local\tsc.js:31813:29)
    at bindSourceFile (D:\github\TypeScript\built\local\tsc.js:30091:17)
    at Object.bindSourceFile (D:\github\TypeScript\built\local\tsc.js:30024:9)
    at initializeTypeChecker (D:\github\TypeScript\built\local\tsc.js:66416:20)
    at Object.createTypeChecker (D:\github\TypeScript\built\local\tsc.js:34831:9)
    at getDiagnosticsProducingTypeChecker (D:\github\TypeScript\built\local\tsc.js:98564:93)
    at Object.getGlobalDiagnostics (D:\github\TypeScript\built\local\tsc.js:98991:72)
    at emitFilesAndReportErrors (D:\github\TypeScript\built\local\tsc.js:102870:53)
```
